include/zephyr/init.h

```diff
@@ -96,7 +107,12 @@ struct init_entry {
 	 * If the init entry belongs to a device, this fields stores a
 	 * reference to it, otherwise it is set to NULL.
 	 */
-	const struct device *dev;
+	union {
+		const struct device *dev;
+#ifdef CONFIG_DEVICE_MUTABLE
+		struct device *dev_rw;
+#endif
+	};
```

Comment:
@marc-hb `__init_enable_logger` is a `SYS_INIT`, I think `unknown field ‘dev’ specified in initializer` is referring to this, and it may need the brace to find the field in the union.

Reply:
was able to repro: https://godbolt.org/z/Whf44q63s does not work on GCC 4, I'll send a PR, easy enough

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/65912